src/expressions/HasLabel.ts

```diff
@@ -33,26 +34,40 @@ import type { NodeRef } from "../references/NodeRef";
  */
 export class HasLabel extends CypherASTNode {
     private node: NodeRef;
-    private expectedLabels: string[];
+    private expectedLabels: string[] | LabelExpr;
 
     /**
      * @hidden
      */
-    constructor(node: NodeRef, expectedLabels: string[]) {
+    constructor(node: NodeRef, expectedLabels: string[] | LabelExpr) {
         super();
-        if (expectedLabels.length === 0) throw new Error("HasLabel needs at least 1 label");
+        this.validateLabelsInput(expectedLabels);
         this.node = node;
         this.expectedLabels = expectedLabels;
     }
 
     /** @internal */
     public getCypher(env: CypherEnvironment): string {
         const nodeId = this.node.getCypher(env);
-        const labelsStr = this.expectedLabels
-            .map((label) => {
-                return `:${escapeLabel(label)}`;
-            })
-            .join("");
+        const labelsStr = this.generateLabelExpressionStr(env);
         return `${nodeId}${labelsStr}`;
     }
+
+    private generateLabelExpressionStr(env: CypherEnvironment): string {
+        if (Array.isArray(this.expectedLabels)) {
+            return this.expectedLabels
+                .map((label) => {
+                    return `:${escapeLabel(label)}`;
+                })
+                .join("");
+        } else {
+            return `:${this.expectedLabels.getCypher(env)}`;
```

Comment:
Not a blocker by any means, but it could be nice to make the `:` a well-named constant so it's clear what it's used for